io.github.milkshiift.GoofCord.yml

```diff
@@ -0,0 +1,77 @@
+app-id: io.github.milkshiift.GoofCord
+
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+
+base: org.electronjs.Electron2.BaseApp
+base-version: '23.08'
+
+command: startgoofcord
+separate-locales: false
+
+finish-args:
+  - --socket=pulseaudio
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --share=network
+  - --share=ipc
+  - --talk-name=org.freedesktop.Notifications # Discord push notifs
+  - --talk-name=org.kde.StatusNotifierWatcher # Tray functionalities on KDE
+  - --device=all # Needed for GPU acceleration and the webcam
+  - --filesystem=xdg-videos:ro
+  - --filesystem=xdg-pictures:ro
+  - --filesystem=xdg-download # This and the above two are used for drag-n-drop, and download managing
+  - --env=XCURSOR_PATH=/run/host/user-share/icons:/run/host/share/icons # This is used to show the correct cursor on Wayland
+
+modules:
+  - name: GoofCord
+    buildsystem: simple
+    build-commands:
+      - chmod +x GoofCord*.AppImage
+      - ./GoofCord*.AppImage --appimage-extract
+      - desktop-file-edit --set-key="Exec" --set-value="startgoofcord" --set-icon=$FLATPAK_ID
+        squashfs-root/goofcord.desktop
+      - install -D squashfs-root/usr/share/icons/hicolor/512x512/apps/goofcord.png /app/share/icons/hicolor/512x512/apps/io.github.milkshiift.GoofCord.png
+      - install -D squashfs-root/goofcord.desktop /app/share/applications/io.github.milkshiift.GoofCord.desktop
+      - install -Dm755 startgoofcord /app/bin/startgoofcord
+      - install -D io.github.milkshiift.GoofCord.metainfo.xml -t /app/share/metainfo/
+      - mv squashfs-root /app/bin/goofcord
+
+    sources:
+      - type: file
+        path: startgoofcord
+
+      - type: file
+        url: https://raw.githubusercontent.com/Milkshiift/GoofCord/main/meta/io.github.milkshiift.GoofCord.metainfo.xml
+        sha512: c681ef71cacd247b05945933b072f806aeca0b07424ef5b7e7f108311582c174b24c8f982bf65c89f004e9cd0d755e5d5bbf9f0d1e6d9771626eec8e698f573e
+        x-checker-data:
+          type: json
+          url: https://api.github.com/repos/Milkshiift/GoofCord/contents/meta/io.github.milkshiift.GoofCord.metainfo.xml
+          version-query: .sha
+          url-query: .download_url
+
+      - type: file
+        url: https://github.com/Milkshiift/GoofCord/releases/latest/download/GoofCord-1.4.1-linux-x86_64.AppImage
```

Comment:
why not rebuilding from source?

Reply:
I used Vesktop's and ArmCord's flatpaks as a base, which unpack an AppImage. Is there a benefit to building from source?

Reply:
Yes if you can, always build from source.

Reply:
GoofCord uses Node >21 and pnpm which seem to not be available in flatpak. Building from source also seems like a mostly pointless over-complication that will make maintaining harder

Reply:
you can also help with that.